flink-core/src/main/java/org/apache/flink/api/java/typeutils/TypeExtractor.java

```diff
@@ -1340,16 +1340,15 @@ private <IN1, IN2, OUT> TypeInformation<OUT> createTypeInfoFromFactory(
             List<Type> typeHierarchy,
             TypeInformation<IN1> in1Type,
             TypeInformation<IN2> in2Type) {
-
-        final List<Type> factoryHierarchy = new ArrayList<>(typeHierarchy);
-        final TypeInfoFactory<? super OUT> factory = getClosestFactory(factoryHierarchy, t);
+        final List<Type> newTypeHierarchy = new ArrayList<>(typeHierarchy);
```

Comment:
nit: it is not a hierarchy it is a list - if we are changing this why not change to typeList and rename the related variables to be Lists not hierarchy - so it is more representative.

Reply:
The `typeHierarchy` also is a list, so I followed the style.
But I think you said is right, let's use `typeHierarchyList` and `newTypeHierarchyList`.

Reply:
After a second thought. I guess the original author use the name typeHierarchy due to the list contains the type information according to the hierarchy.
Such as:
{class java.lang.String, class java.lang.Object}.
So let me temporarily use the name `typeHierarchy`. I am willing to use `typeHierarchyList` and `newTypeHierarchyList` if more reviewer suggest it.